cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -386,17 +391,165 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
                 .build();
     }
 
-    private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalanceState cruiseControlState, Map<String, Object> optimizationResult) {
+    /**
+     * Converts the supplied JSONArray containing the load information JSONObject for each broker, into a map linking from
+     * broker ID to a map linking a more readable version of the load parameters key to their values. The load parameters
+     * that are extracted and the readable versions of the keys are dictated by the values defined in the
+     * {@link CruiseControlLoadParameters} enum.
+     *
+     * @param brokerLoadArray The JSONArray of broker load JSONObjects returned by the Cruise Control rebalance endpoint.
+     * @return A map linking from broker ID integer to a map of load parameter to value.
+     */
+    protected static Map<Integer, Map<String, Object>> extractLoadParameters(JsonArray brokerLoadArray) {
+
+        Map<Integer, Map<String, Object>> tempMap = new HashMap<>();
```

Comment:
Does this still need to be called a `tempMap`?

Reply:
A good point, naming things is hard.